### DISCUSSION TITLE
Any recommended fine-tuning settings on Meshcentral such as hearbeat to keep the session alive for longer duration, compression, framerate changes, etc.

### DISCUSSION THREAD

**Discussion by mdshoaibumer** [Ylianst/MeshCentral#3626] [General]
Dear @Ylianst ,

Any recommended fine-tuning settings on Meshcentral such as hearbeat to keep the session alive for longer duration, compression, framerate changes, etc.

**Comment by JSuenram** [comment]
Hi,
from what I can say, the default are pretty much good for nearly all scenarios.

If you Publish MeshCentral with a Reverse-Proxy like haProxy it should have the timeouts set larger then MeshCentral itself to keep sessions alive.

You can always lower Quality and Framerate if you have a specific session on a slow long distance line. 
When you have multiple Admins connecting to the same remote session consider enabling the desktopMultiplex-Setting in config.
This saves some bandwidth. 

But if you have currently no problems, leave settings at default. Don't repair when it is not broken.... ;-)

**Reply by timcanty** [reply]
Hi @JSuenram In the haproxy situation you mentioned the timeouts should be set higher then the settings in meshcentral, 
do we now what the standard timeouts are set as in meshcentral, so that we can make this change in haproxy? Only recently started using meshcentral, so might be missing where these defaults are documented,

**Reply by JSuenram** [reply]
You can have a look on page 30 in den Userguide. https://info.meshcentral.com/downloads/MeshCentral2/MeshCentral2UserGuide-0.2.6.pdf It is for NGINX but can be adopted to HAPROXY easy....

> In this example, we will:
 MeshCentral on non-standard ports, but alias HTTPS to port 443.
 NGINX will be using standard ports 80 and 443.
 We will have NGINX perform all TLS authentication & encryption.
 MeshCentral will read the NGINX web certificate so agents will perform correct server
authentication.
 NGINX will be setup with long timeouts, because agents have long standard web socket
connections.
Let’s get started by configuring MeshCentral with the following values in config.json:
       {
         "settings": {
           "Cert": "myservername.domain.com"
           "Port": 4430,
           "AliasPort": 443,
           "RedirPort": 800,
           "AgentPong": 300,
           "TlsOffload": "127.0.0.1"
         },
         "domains": {
           "": {
             "certUrl": "https://127.0.0.1:443/"
           }
} }
With this configuration, MeshCentral will be using port 4430 instead of port 443, but because “TlsOffload” is set, TLS will not be performed on port 4430. The server name is set to “myservername.domain.com”, so that is the name that MeshCentral will give to agents to connect to. Also, the alias port is set to 443. So agents will be told to connect to “myservername.domain.com:443”.
The “AgentPong” line instructs the server to send data to the agent each 300 seconds and the agent by default will send data to the server every 120 seconds. As long as NGINX timeouts are longer than this, connections should remain open.

**Comment by JSuenram** [comment]
@mdshoaibumer 
If your question is answered, pls mark as answered.
Thank you!

**Comment by Ylianst** [comment]
I have a video on [reducing server hosting cost with compression](https://www.youtube.com/watch?v=o_RuYyoJj0g). Lots of tips and tricks here.

As for the heartbeat, [take a look at my answer here](https://github.com/Ylianst/MeshCentral/discussions/3672).

**Reply by PrplHaz4** [reply]
> I have a video on [reducing server hosting cost with compression](https://www.youtube.com/watch?v=o_RuYyoJj0g). Lots of tips and tricks here.

For reference:
```json
"settings": {
    ...
    "compression": true,
    "wsCompression": false,
    "agentWsCompression": true,
    "webRTC": true
  },
  "domains": {
    "": {
      ...
      "minify": true,
      ...
    }
  },
